src/app/settings/dimSettingsService.factory.js

```diff
@@ -110,12 +103,16 @@ export function SettingsService($rootScope, SyncService, $window, $i18next, $q)
     _ready.resolve();
 
     $rootScope.$evalAsync(() => {
+      const languageChanged = savedSettings.language !== i18next.language;
       angular.merge(settings, savedSettings);
-      changeLanguage(settings.language, () => {
-        $rootScope.$applyAsync(() => {
-          $rootScope.$broadcast('i18nextLanguageChange');
+      localStorage.dimLanguage = settings.language;
+      if (languageChanged) {
+        changeLanguage(settings.language, () => {
```

Comment:
nice, so we still don't need to force a full refresh to load a new language?

Reply:
Nope, I fixed that when I redid settings. You still need to reload for item translations to update, but we show a handy button for that.